### ISSUE TITLE
Symfony version 3.4.20 causes infinite loop while loading translations

### ISSUE THREAD

**Issue by vytsci** [oroinc/platform#912]
OroLocale uses Symfony INTL and version fixed within you composer is causing infinite loop as it has bug. Bug appears in vendor/symfony/symfony/src/Symfony/Component/Intl/Data/Bundle/Reader/BundleEntryReader.php file on 81 line where $currentLocale will never be null it should be false. I dont know why version is fixed this way, but it needs to be changed.

**Comment by vytsci** [comment]
Solution was to override symfony file

"autoload": {
    ...
    "files": [
      ...
      "src/FixMe/Symfony/Component/Intl/Data/Bundle/Reader/BundleEntryReader.php"
    ],
    ...
  },

and on line 81 replace `while (null !== $currentLocale) {` with `while (false !== $currentLocale) {`

This will make the cycle break and cache now works again.

**Comment by vytsci** [comment]
The bug occurs after new php intl update. Somehow it returns wrong data. On my laptop where php intl is older everything works and on my work PC I need this workaround.

**Comment by vtsykun** [comment]
Fixed in symfony/symfony 3.4.24 https://github.com/symfony/symfony/pull/30781.

libicu 64.1 is responsible for this change. After updating php version 7.1.26 => 7.1.28 it uses another version libicu http://site.icu-project.org/download/64. 
```
Migration Issues
The locale ID handling for "", "und" and "root" has been made consistent across Java, 
C++, and C. In particular, Locale.getLanguage() will return the empty string for each 
of these now, Locale("und_IN").getName() will be "_IN" for consistency, etc. 
This might require some adjustments.
```
**php 7.1.28**
```
intl
ICU version => 64.1
```

**php 7.1.26**
```
intl
ICU version => 60.2
```

To fix you can apply patch 
```patch
diff --git a/src/Symfony/Component/Intl/Locale.php b/src/Symfony/Component/Intl/Locale.php
index a9c36ec618b5..6e88c5629e00 100644
--- a/src/Symfony/Component/Intl/Locale.php
+++ b/src/Symfony/Component/Intl/Locale.php
@@ -85,7 +85,9 @@ public static function getFallback($locale)
 
             array_pop($localeSubTags);
 
-            return locale_compose($localeSubTags);
+            $fallback = locale_compose($localeSubTags);
+
+            return false !== $fallback ? $fallback : null;
         }
 
         if (false !== $pos = strrpos($locale, '_')) {
```

**Comment by vytsci** [comment]
OK thank you. Strange issue.

**Comment by anyt** [comment]
Internal ticket id #BAP-18262

**Comment by SGPaul** [comment]
Symfony 2.8 has same issue? Should i open new ticket? (Or is this no longer supported for 2.8?)

**Comment by klaussilveira** [comment]
Any news on this?

**Comment by aivus** [comment]
@klaussilveira it was fixed in platform 3.1.8

**Comment by klaussilveira** [comment]
Fixed in: https://github.com/oroinc/platform/commit/75efd99cb9faff5a7f84790aaad2e2fea16dd679

This should have been included in the changelog, given the importance of this bug. Can you close the ticket then? Thanks!
